File: Recursion_Array/display.cpp
#include<iostream>
#include<vector>
using namespace std;

void display(vector<int>& arr, int idx)
{
    if(idx == arr.size())
    {
        return;
    }
    
    cout << arr[idx] << endl;
    display(arr, idx + 1);
}

void input(vector<int>& arr)
{
    for(int i = 0; i < arr.size(); i++)
    {
        cin >> arr[i];
    }
}

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    input(arr);
    display(arr, 0);
}

File: Recursion_Array/display.java
import java.util.Scanner;
public class display
{
    public static Scanner scn = new Scanner(System.in);
    public static void display(int[] arr, int idx)
    {
        if(idx == arr.length)
        {
            return;
        }
        System.out.println(arr[idx]);
        display(arr, idx + 1);
    }

    public static void input(int[] arr)
    {
        for(int i = 0; i < arr.length; i++)
        {
            arr[i] = scn.nextInt();
        }
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        display(arr, 0);
    }
}